File: frontend/src/stores/category.ts
import { createCrudStore } from './createCrudStore'

export interface Category {
  id: number
  name: string
  icon: string
  color: string
  display_order: number
  transaction_count: number
  total_amount: number
}

export const useCategoryStore = createCrudStore<Category>({
  name: 'category',
  endpoint: '/v1/categories',
})

File: backend/models/category.py
from sqlalchemy import ForeignKey, Integer, String
from sqlalchemy.orm import Mapped, mapped_column

from database import Base
from models.mixins import TimestampMixin


class Category(Base, TimestampMixin):
    __tablename__ = "category"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    uid: Mapped[int] = mapped_column(Integer, ForeignKey("user.id"), nullable=False)
    name: Mapped[str] = mapped_column(String(32), nullable=False)
    icon: Mapped[str] = mapped_column(String(10), nullable=False, default="📦")
    color: Mapped[str] = mapped_column(String(7), nullable=False, default="#607D8B")
    display_order: Mapped[int] = mapped_column(Integer, nullable=False, default=0)
